charts/gateway-helm/templates/envoy-proxy-topology-mutator-webhook.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.topologyWebhook.enabled }}
+{{- $secret := default (dict "data" (dict)) (lookup "v1" "Secret" .Release.Namespace "envoy-gateway") }}
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: envoy-gateway-topology-webhook
+  {{- with .Values.topologyWebhook.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    app.kubernetes.io/component: topology-webhook
+  {{- include "eg.labels" . | nindent 4 }}
+webhooks:
+  - name: topology.webhook.envoyproxy.io
+    admissionReviewVersions: ["v1"]
+    sideEffects: NoneOnDryRun
+    timeoutSeconds: 5
+    clientConfig:
+      caBundle: {{ default "" (index $secret.data "ca.crt" ) | b64dec }}
```

Comment:
I'll clean this up a bit more but we can't use the internal certgen because we need the ca cert data at deploy time. An alternative here would be to conditionally use cert-manager ca cert injection which is what a lot of other charts do.